App crash on startup on some devices without Google services

```
java.lang.RuntimeException: Unable to create application org.wordpress.android.WordPress: java.lang.SecurityException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.REGISTER pkg=com.google.android.gsf (has extras) } without permission com.google.android.c2dm.permission.RECEIVE
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4362)
       at android.app.ActivityThread.access$1500(ActivityThread.java:136)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1279)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5032)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
Caused by: java.lang.SecurityException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.REGISTER pkg=com.google.android.gsf (has extras) } without permission com.google.android.c2dm.permission.RECEIVE
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1555)
       at android.app.ContextImpl.startService(ContextImpl.java:1532)
       at android.content.ContextWrapper.startService(ContextWrapper.java:494)
       at com.google.android.gcm.GCMRegistrar.internalRegister(GCMRegistrar.java:230)
       at com.google.android.gcm.GCMRegistrar.register(GCMRegistrar.java:218)
       at org.wordpress.android.WordPress.gcmRegisterIfNot(WordPress.java:342)
       at org.wordpress.android.WordPress.registerForCloudMessaging(WordPress.java:357)
       at org.wordpress.android.WordPress.onCreate(WordPress.java:215)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1014)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4359)
       at android.app.ActivityThread.access$1500(ActivityThread.java:136)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1279)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:136)
       at android.app.ActivityThread.main(ActivityThread.java:5032)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
``
```

--- a/WordPress/src/main/java/org/wordpress/android/WordPress.java
+++ b/WordPress/src/main/java/org/wordpress/android/WordPress.java
@@ -343,6 +343,10 @@ public class WordPress extends Application {
             // GCMRegistrar.checkManifest or GCMRegistrar.register throws an IllegalStateException if Manifest
             // configuration is incorrect (missing a permission for instance) or if GCM dependencies are missing
             AppLog.e(T.NOTIFS, "APK (manifest error or dependency missing) doesn't support GCM: " + e.getMessage());
+        } catch (Exception e) {
+            // SecurityException can happen on some devices without Google services (these devices probably strip
+            // the AndroidManifest.xml and remove unsupported permissions).
+            AppLog.e(T.NOTIFS, e);
         }
         return regId;
     }

